packages/driver/src/cypress/cy.ts

```diff
@@ -1257,11 +1259,40 @@ export class $Cy extends EventEmitter2 implements ITimeouts, IStability, IAssert
 
     args.unshift(subject)
 
-    this.Cypress.action('cy:next:subject:prepared', subject, args, firstCall)
-
     return args
   }
 
+  subjectForChainer (chainerId: string) {
+    return (this.state('subject') || {})[chainerId]
+  }
+
+  linkSubject (fromChainerId, toChainerId) {
+    const links = this.state('subjectLinks') || {}
+
+    links[fromChainerId] = toChainerId
+    this.state('subjectLinks', links)
+  }
+
+  breakSubjectLinksToCurrentChainer () {
```

Comment:
Thoughts on adding in some docs on what these are doing? Seems like a lot of code to replace `.state('subject')`

Curious on what rules/scenarios are  `breakSubjectLinksToCurrentChainer` for should be used?

Reply:
Ok, so that novel of a comment aside, I'm unsure if I want to document this *right now* - I believe the next step (introducing the concept of 'selectors' and subject chains) will change how this works again, and commenting the intermediate stage extensively didn't feel quite right.

With that said, I'm not exactly sure how much will change - it may be worth documenting the current state in-code, rather than just in my head or in PR comments like this. I'll add a comment here (but probably not a complete description of the system).

Reply:
Nevermind, I started writing and then words came out. Many lines of comments added around the new cy functions.

Reply:
These comments are amazing! 🤩